Classifier and Summarizer/WebEntityDetection.py

```
@@ -0,0 +1,120 @@
+          }
+      }
+    }
+
+  def prepareRequests(self,encodedImages:list) ->list:
+    '''
+    prepares the requests for making calls
+    to the api. Structures the given images
+    in the format as expected by the API
+    Params :
+      encodedImages : numpy.ndarray of base64 encoded images
+    Return Type:
+    list: of required format
+    '''
+    imageRequests = []
+    
+    for image in encodedImages:
+      imageRequests.append(self.formatSingleRequest(image))
+    
+    return imageRequests
+
+
+  def getBestGuessLabel(self,jsonResponse,imageIndex):
+    ''' return the best guess label '''
+    return jsonResponse["responses"][imageIndex]["webDetection"]["bestGuessLabels"][0]["label"]
+
+  def getTopEntities(self,jsonResponse,numberOfEntities,imageIndex):
+    ''' return the top entity description'''
+    topEntites = ""
+    numberOfEntities = min(numberOfEntities, len(jsonResponse["responses"][imageIndex]["webDetection"]["webEntities"]))
+    for i in range(numberOfEntities):
+      if "description" in jsonResponse["responses"][imageIndex]["webDetection"]["webEntities"][i]:
+        topEntites+=jsonResponse["responses"][imageIndex]["webDetection"]["webEntities"][i]["description"]+"."
+    return topEntites
+
+
+  def getDescriptionForImages(self,images:list) -> list:
+    '''
+    given a list of images returns the best guess
+    labels for the web entities or description of
+    the images
+    Params:
+      images : list of image paths
+    Return type:
+    list<str> : list containing best guess descriptions of the image
+    '''
+    base64EncodedImages = self.findBase64EncodingForImages(images)
+
+    imageRequests = self.prepareRequests(base64EncodedImages)
+    
+    numberOfImages = len(images)
+    
+    jsonDataForPostRequest = json.dumps({
+      "requests": imageRequests
+    })
+    response = requests.post(self.apiUrl,data=jsonDataForPostRequest)
+    if response.status_code!=200:
+      return []
```

Can we return a better error description/indicator here?